python/tests/test_highlevel.py

```diff
@@ -2027,8 +2044,21 @@ def test_str(self, ts):
         assert len(s) > 999
         assert re.search(rf"║Trees *│ *{ts.num_trees}║", s)
         assert re.search(rf"║Time Units *│ *{ts.time_units}║", s)
-        for table in ts.tables.table_name_map:
-            assert re.search(rf"║{table.capitalize()} *│", s)
+        codecs = collections.defaultdict(int)
```

Comment:
I'm very much negative on this style of testing now - any complex logic in a test case is to be avoided.

I suggested testing these functions as follows

```
class TestMetadataCodecStr:
     def test_plain_table_collection(self):
             s = util.metadata_codec(tskit.TableCollection(1))
             assert s == "raw"

```

This is much more useful long term as well as being easier to write and review.